schemes/sima_diagnostics/zm_diagnostics.F90

```diff
@@ -0,0 +1,206 @@
+                            'lev',  'avg', 'm s-2')
+
+    call history_add_field ('ZMICUU', 'in_cloud_eastward_wind_in_updraft_due_to_deep_convection',  'lev',  'avg', 'm s-1')
+    call history_add_field ('ZMICUD', 'in_cloud_eastward_wind_in_downdraft_due_to_deep_convection',  'lev',  'avg', 'm s-1')
+    call history_add_field ('ZMICVU', 'in_cloud_northward_wind_in_updraft_due_to_deep_convection',  'lev',  'avg', 'm s-1')
+    call history_add_field ('ZMICVD', 'in_cloud_northward_wind_in_downdraft_due_to_deep_convection', 'lev',  'avg', 'm s-1')
+
+    call history_add_field ('DLFZM',   'detrainment_of_cloud_liquid_due_to_deep_convection', 'lev', 'avg','kg kg-1 s-1 ')
+
+
+   end subroutine zm_diagnostics_init
+
+   !> \section arg_table_zm_diagnostics_run  Argument Table
+   !! \htmlinclude zm_diagnostics_run.html
+   subroutine zm_diagnostics_run(ncol, pver, pverp, ideep, cpair, prec, cape, gravit, mu, md, &
+      dlf, ps, pap, maxg, jt, flxprec, flxsnow, ntprprd, ntsnprd, pguallu, pguallv, &
+      pgdallu, pgdallv, icwuu, icwuv, icwdu, icwdv, mcon, errmsg, errflg)
+
+      use cam_history, only: history_out_field
+
+      !------------------------------------------------
+      !   Input / output parameters
+      !------------------------------------------------
+      integer, intent(in) :: ncol
+      integer, intent(in) :: pver
+      integer, intent(in) :: pverp
+      integer, intent(in) :: ideep(:)
+
+      real(kind_phys), intent(in) :: cpair
+      real(kind_phys), intent(in) :: prec(:)
+      real(kind_phys), intent(in) :: cape(:)
+      real(kind_phys), intent(in) :: gravit
+      real(kind_phys), intent(in) :: mu(:,:)
+      real(kind_phys), intent(in) :: md(:,:)
+      real(kind_phys), intent(in) :: dlf(:,:)
+      real(kind_phys), intent(in) :: ps(:)
+      real(kind_phys), intent(in) :: pap(:,:)
+      integer, intent(in) :: maxg(ncol)
+      integer, intent(in) :: jt(ncol)
+      real(kind_phys),intent(in) :: flxprec(:,:)
+      real(kind_phys),intent(in) :: flxsnow(:,:)
+      real(kind_phys),intent(in) :: ntprprd(:,:)
+      real(kind_phys),intent(in) :: ntsnprd(:,:)
+      real(kind_phys),intent(in) :: pguallu(:,:)
+      real(kind_phys),intent(in) :: pguallv(:,:)
+      real(kind_phys),intent(in) :: pgdallu(:,:)
+      real(kind_phys),intent(in) :: pgdallv(:,:)
+      real(kind_phys),intent(in) :: icwuu(:,:)
+      real(kind_phys),intent(in) :: icwuv(:,:)
+      real(kind_phys),intent(in) :: icwdu(:,:)
+      real(kind_phys),intent(in) :: icwdv(:,:)
+
+      real(kind_phys),intent(inout) :: mcon(:,:)
+
+      ! CCPP error handling variables
+      character(len=512), intent(out) :: errmsg
+      integer,          intent(out) :: errflg
+
+      integer :: lengath  ! number of columns with deep convection
+
+      real(kind_phys) :: freqzm(ncol)
+      real(kind_phys) :: pcont(ncol)
+      real(kind_phys) :: pconb(ncol)
+      real(kind_phys) :: ftem(ncol,pver)
+      real(kind_phys) :: mconzm(ncol,pverp)
```

Comment:
This variable (`mconzm`) is no longer used.

Reply:
removed